src/sentry/grouping/enhancer/matchers.py

```diff
@@ -59,34 +61,60 @@ def _get_function_name(frame_data: dict, platform: Optional[str]):
 
 def create_match_frame(frame_data: dict, platform: Optional[str]) -> dict:
     """ Create flat dict of values relevant to matchers """
-    match_frame = dict(
-        category=get_path(frame_data, "data", "category"),
-        family=get_behavior_family_for_platform(frame_data.get("platform") or platform),
-        function=_get_function_name(frame_data, platform),
-        in_app=frame_data.get("in_app"),
-        module=get_path(frame_data, "module"),
-        package=frame_data.get("package"),
-        path=frame_data.get("abs_path") or frame_data.get("filename"),
-    )
+    return frame_data
 
-    for key in list(match_frame.keys()):
-        value = match_frame[key]
-        if isinstance(value, (bytes, str)):
-            if key in ("package", "path"):
-                value = match_frame[key] = value.lower()
 
-            if isinstance(value, str):
-                match_frame[key] = value.encode("utf-8")
+@dataclass
+class MatchConfig:
+    field: str
+    path_like: bool
+    pattern_is_glob: bool
+    negated: bool
+    patterns: Sequence[str]
+    required: bool = False
 
-    return match_frame
 
-
-class Match:
+class Match(ABC):
     description = None
 
+    @abstractmethod
+    def extract_frame_data(self, frames, idx, platform, exception_data):
+        raise NotImplementedError()
+
     def matches_frame(self, frames, idx, platform, exception_data, cache):
+        config = self.get_match_config()
+        global_extracted, frame_extracted = self.extract_frame_data(
+            frames, idx, platform, exception_data
+        )
+        if config.field in frame_extracted:
+            val = frame_extracted[config.field]
+        elif config.field in global_extracted:
+            val = global_extracted[config.field]
+        elif config.required:
+            return False
+        else:
+            val = None
+
+        if val is None:
+            is_match = False
+        elif config.pattern_is_glob:
+            if config.path_like:
+                is_match = any(path_like_match(val, pat.encode("utf8")) for pat in config.patterns)
```

Comment:
Wouldn't it still be better to encode the pattern in the matcher's constructor, so it doesn't have to be repeated on every frame?